samples/robot/src/main/kotlin/example/robot.kt

```diff
@@ -2,23 +2,38 @@ package example
 
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
 import com.jakewharton.mosaic.Column
+import com.jakewharton.mosaic.KeyEvent
+import com.jakewharton.mosaic.Terminal
 import com.jakewharton.mosaic.Text
+import com.jakewharton.mosaic.onKeyEvent
 import com.jakewharton.mosaic.runMosaic
-import kotlinx.coroutines.Dispatchers.IO
-import kotlinx.coroutines.withContext
-import org.jline.terminal.TerminalBuilder
-
-private const val width = 20
-private const val height = 10
+import kotlinx.coroutines.CompletableDeferred
 
 fun main() = runMosaic {
-	// TODO https://github.com/JakeWharton/mosaic/issues/3
-	var x by mutableStateOf(0)
-	var y by mutableStateOf(0)
+	val exitSignal = CompletableDeferred<Unit>()
 
 	setContent {
+		var x by remember { mutableStateOf(0) }
+		var y by remember { mutableStateOf(0) }
+		val width = Terminal.current.size.width
+
+		// 3 for header stuff, 1 for final newline, 1 for cursor line.
+		val height = Terminal.current.size.height - 5
+
+		onKeyEvent { event ->
+			when (event) {
+				KeyEvent.Q -> exitSignal.complete(Unit)
+				KeyEvent.UP -> y = (y - 1).coerceAtLeast(0)
+				KeyEvent.DOWN -> y = (y + 1).coerceAtMost(height)
+				KeyEvent.RIGHT -> x = (x + 1).coerceAtMost(width)
```

Comment:
```suggestion
				KeyEvent.RIGHT -> x = (x + 1).coerceAtMost(width - robot.width - 1)
```
otherwise it'll try to render characters off-screen on a 120 terminal 116 is the max valid x.

![image](https://user-images.githubusercontent.com/2906988/200536807-8ae6d569-48ca-4332-bb12-e106d59d55fd.png)